src/metrics-gatherer.ts

```diff
@@ -134,81 +193,141 @@ export class MetricsGatherer {
+				return;
+			}
+			// if already exists with another kind, throw error
+			if (this.kinds[name] && this.kinds[name] !== kind) {
+				this.err(
+					new MetricsGathererError(
+						`tried to use ${name} twice - first as ` +
+							`${this.kinds[name]}, then as ${kind}`,
+					),
+				);
+				return;
+			}
+			// if not described, describe (poorly)
+			if (!(name in this.descriptions)) {
+				this.descriptions[name] = `undescribed ${kind} metric`;
+			}
+			// if doesn't exist, create metric
+			this.metrics[kind][name] = constructors[kind].create({
+				name,
+				help: this.descriptions[name],
+				labelNames: Object.keys(labels),
+				...custom,
+				...this.customParams[name],
+			});
+			this.kinds[name] = kind;
+			this.existMap[name] = true;
+		} catch (e) {
+			console.error('caught ' + e);
+			this.err(e);
 		}
-		// if doesn't exist, create metric
-		this.metrics[kind][name] = constructors[kind].create({
-			name,
-			help: this.descriptions[name],
-			labelNames: Object.keys(labels),
-			...custom,
-			...this.customParams[name],
-		});
-		this.kinds[name] = kind;
-		this.existMap[name] = true;
 	}
 
 	// reset a given metric by name
 	public reset(name: string) {
-		// will only have an entry in this.kinds if the metric has been observed
-		// at least once (don't need to "reset" otherwise, and avoids error in
-		// indexing)
-		if (this.kinds[name]) {
-			this.metrics[this.kinds[name]][name].reset();
+		try {
+			// will only have an entry in this.kinds if the metric has been observed
+			// at least once (don't need to "reset" otherwise, and avoids error in
+			// indexing)
+			if (this.kinds[name]) {
+				this.metrics[this.kinds[name]][name].reset();
+			}
+		} catch (e) {
+			console.error('caught ' + e);
+			this.err(e);
 		}
 	}
 
 	// create an express request handler given an auth test function
 	public requestHandler(authTest?: AuthTestFunc): express.Handler {
-		return (req: express.Request, res: express.Response) => {
-			if (authTest && !authTest(req)) {
-				return res.status(403).send();
-			}
-			res.writeHead(200, { 'Content-Type': 'text/plain' });
-			res.end(prometheus.register.metrics());
+		let handler: express.Handler = (
+			_req: express.Request,
+			_res: express.Response,
+		) => {
+			console.error(
+				'error in metrics.requestHandler() caused empty express metrics request handler to be created',
+			);
 		};
+		try {
+			handler = (req: express.Request, res: express.Response) => {
+				if (authTest && !authTest(req)) {
+					return res.status(403).send();
+				}
+				res.writeHead(200, { 'Content-Type': 'text/plain' });
+				res.end(prometheus.register.metrics());
+			};
+		} catch (e) {
+			console.error('caught ' + e);
+			this.err(e);
+		}
+		return handler;
 	}
 
+	// create an express request handler given an auth test function which is
+	// suitable for use in a context where we're using node's `cluster` module
 	public aggregateRequestHandler(authTest?: AuthTestFunc): express.Handler {
```

Comment:
This is a one-off setup function right?  If so then it probably makes sense that this does actually throw an error on failure